File: dev-docs/snippets/events/document-opened.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Fires when a document is opened (subscribed to) by this client.
export function documentOpened(client: JsBaoClient) {
  // #region example
  client.on("documentOpened", (payload) => {
    console.log("opened", payload.documentId);
  });
  // #endregion example
}

File: dev-docs/snippets/events/document-opened.swift
import JsBaoClient

// Fires when a document is opened (subscribed to) by this client.
func documentOpened(client: JsBaoClient) {
  // #region example
  let sub = client.events.on(.documentOpened) { (event: DocumentOpenedEvent) in
    print("opened", event.documentId)
  }
  // #endregion example
  _ = sub
}
